### PR TITLE
External controller log message

### PR DESCRIPTION
**Pull request by SimonRichardson**
Whilst debugging a production issue, we saw the log line that controller info was updated, but we don't know what it went from and to. As this happened a lot, it would be helpful to better understand why we're constantly getting a new set of addresses.

Neither the cert or the alias are updated when we get new information, so it doesn't make sense to keep spitting them out. Just the addresses are enough to get an idea of the churn.

## Checklist

- [x] Code style: imports ordered, good names, simple structure, etc
- [x] Comments saying why design decisions were made
- [x] Go unit tests, with comments saying what you're testing

## QA steps

```
$ juju bootstrap lxd test
$ juju add-model default
$ juju deploy ubuntu
$ juju enable-ha
```

You should see the line about the controller info being updated.

## Links

**Jira card:** JUJU-

### CODE REVIEW

File: worker/externalcontrollerupdater/externalcontrollerupdater.go [hunk 1]
```diff
@@ -247,20 +247,26 @@ func (w *controllerWatcher) loop() error {
...
 				Alias:         info.Alias,
-				Addrs:         info.Addrs,
+				Addrs:         newInfo.Addrs,
 				CACert:        info.CACert,
 			}); err != nil {
 				return errors.Annotate(err, "caching external controller info")
 			}
-			logger.Infof("new controller info for controller %q: %v", w.tag.Id(), info)
+
+			logger.Infof("new controller info for controller %q: addresses changed: new %v, prev %v", w.tag.Id(), newInfo.Addrs, info.Addrs)
```
Comment (ukane-philemon) [inline worker/externalcontrollerupdater/externalcontrollerupdater.go:264]: The previous log message logged `info`, is there any reason we don't need to log the rest information there?

**Review by ukane-philemon — APPROVED** [review]
LGTM

File: worker/externalcontrollerupdater/externalcontrollerupdater.go (hunk 1 above)
Comment (SimonRichardson) [reply worker/externalcontrollerupdater/externalcontrollerupdater.go:264]: It never changed, so it just added a lot of noise to the logs. Especially the cert.

**Review by hpidcock — APPROVED** [review]

**Pull request merged**